src/Microsoft.Extensions.DiagnosticAdapter/Internal/ProxyAssembly.cs

```diff
@@ -41,7 +41,7 @@ static ProxyAssembly()
             return ModuleBuilder.DefineType(name, attributes, baseType, interfaces);
         }
 
-#if GENERATE_ASSEMBLIES
+#if NET451 && GENERATE_ASSEMBLIES
```

Comment:
???

Obvi this functionality isn't available on CoreCLR. I solved this in the past by just defining this for net451